libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -603,6 +606,20 @@ class MapboxNavigation(
         tripSession.unregisterRouteAlertsObserver(routeAlertsObserver)
     }
 
+    /**
+     * Observer will be called when the EHorizon changes.
+     */
+    fun registerEHorizonObserver(eHorizonObserver: EHorizonObserver) {
+        tripSession.registerEHorizonObserver(eHorizonObserver)
+    }
```

Comment:
We should immediately notify the observer with the latest.

Reply:
Good question. I guess due to reactive approach. What is immediate?

Reply:
Thought about it and it's tricky because we don't handle the state, it's in NN side.

Reply:
@LukasPaczos Check out https://github.com/mapbox/mapbox-navigation-android/blob/7408f4958412e4fc2757e5cb8b8631d34d7a06a3/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L423-L430 What do you think?